assets/model_monitoring/components/src/token_statistics_compute_metrics/run.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Annotation Metrics Computing Spark Component."""
+
+import argparse
+from pyspark.sql.functions import lit
+from pyspark.context import SparkContext
+from pyspark.sql.session import SparkSession
+from shared_utilities.io_utils import (
+    read_mltable_in_spark,
+    save_spark_df_as_mltable,
+    init_spark,
+)
+from compute_token_statistics_metrics import *
+
+
+def run():
+    # Init spark session
+    spark = init_spark()
+
+    # Parse argument
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--token_dataset", type=str)
+    parser.add_argument("--group_pivot_column_name", type=str, required=False, nargs="?")
+    parser.add_argument("--signal_metrics", type=str)
+
+    args = parser.parse_args()
+    token_df = read_mltable_in_spark(args.token_dataset)
+    group_pivot_column_name = args.group_pivot_column_name
+
+    # failed calls dont have an id, so imputing them
+    token_df = impute_ids_for_failed_calls(token_df)
+
+    # basic data quality checks and filter out rows that dont meet the quality criteria
+    token_df = check_data_quality(token_df)
+
+    # designate the group column
+    token_df = token_df.withColumnRenamed("node_id", "group")
+
+    #  designate the group_pivot column 
+    #  if group_pivot_column_name is null, then add a column with value Not Provided
+    if group_pivot_column_name is None:
+        token_df = token_df.withColumn("group_pivot", lit('').cast("string"))
+    else:
+        token_df = token_df.withColumnRenamed(group_pivot_column_name, "group_pivot")
+
+    # total tokens = prompt tokens + completion tokens
+    token_df = token_df.withColumn("total_tokens",\
+                                    token_df["prompt_tokens"] + token_df["completion_tokens"])
+
+
+    # compute GPU utilization metrics for group
+    ## create a copy of token_df where group_pivot is set to Aggregate
+    token_df_aggregate_group_pivot = token_df.withColumn("group_pivot", lit('Aggregate').cast("string"))
+    ## compute GPU utilization metrics for group
+    gpu_utilization_metrics_group_only = compute_GPU_utilization_metrics(token_df_aggregate_group_pivot)
```

Comment:
nit: in theory this should be `compute_GPU_utilization_metrics(token_df)` I think, as the metric compute logic doesn't require the "aggregate copy".

Reply:
Now I see it. This is for group only.